compiler/semcall.nim

```diff
@@ -482,7 +482,10 @@ proc inferWithMetatype(c: PContext, formal: PType,
     # This almost exactly replicates the steps taken by the compiler during
     # param matching. It performs an embarrassing amount of back-and-forth
     # type jugling, but it's the price to pay for consistency and correctness
-    result.typ = generateTypeInstance(c, m.bindings, arg.info,
+    if formal.kind == tyUserTypeClass:
+      result.typ = arg.typ
```

Comment:
I didn't see where concept types are resolved to concrete types. When it seemed to be a `tyConcept` type at the c code gen phase, not a string type. It causes difficulties for the ORC code generator, which initializes the string with a `tyConcept` type using `static const` which should have been a variable.